cedar-policy-validator/src/types.rs

```diff
@@ -689,10 +703,12 @@ impl Display for Type {
                 write!(f, "{{")?;
                 for (name, ty) in attrs.iter() {
                     write!(f, "{name}")?;
-                    if !ty.is_required {
+                    if !ty.is_required() {
                         write!(f, "?")?;
                     }
-                    write!(f, ": {},", ty.attr_type)?;
+                    write!(f, ": ")?;
+                    ty.display(f)?;
```

Comment:
Could rename this function to `display_type` or something so it's clear this doesn't re-print  the optional qualifier.